presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -92,101 +92,223 @@ public class PrestoNativeQueryRunnerUtils
 
     private PrestoNativeQueryRunnerUtils() {}
 
-    public static QueryRunner createQueryRunner(boolean addStorageFormatToPath, boolean isCoordinatorSidecarEnabled, boolean enableRuntimeMetricsCollection, boolean enableSsdCache)
-            throws Exception
+    public static Builder builder()
     {
-        int cacheMaxSize = 4096; // 4GB size cache
-        NativeQueryRunnerParameters nativeQueryRunnerParameters = getNativeQueryRunnerParameters();
-        return createQueryRunner(
-                Optional.of(nativeQueryRunnerParameters.serverBinary.toString()),
-                Optional.of(nativeQueryRunnerParameters.dataDirectory),
-                nativeQueryRunnerParameters.workerCount,
-                cacheMaxSize,
-                DEFAULT_STORAGE_FORMAT,
-                addStorageFormatToPath,
-                isCoordinatorSidecarEnabled,
-                enableRuntimeMetricsCollection,
-                enableSsdCache);
+        return new Builder();
     }
 
-    public static QueryRunner createQueryRunner(
-            Optional<String> prestoServerPath,
-            Optional<Path> dataDirectory,
-            Optional<Integer> workerCount,
-            int cacheMaxSize,
-            String storageFormat,
-            boolean addStorageFormatToPath,
-            boolean isCoordinatorSidecarEnabled,
-            boolean enableRuntimeMetricsCollection,
-            boolean enableSsdCache)
-            throws Exception
+    public static class Builder
     {
-        QueryRunner defaultQueryRunner = createJavaQueryRunner(dataDirectory, storageFormat, addStorageFormatToPath);
+        private Builder()
+        {
+            NativeQueryRunnerParameters nativeQueryRunnerParameters = getNativeQueryRunnerParameters();
+            dataDirectory = nativeQueryRunnerParameters.dataDirectory;
+            serverBinary = nativeQueryRunnerParameters.serverBinary.toString();
+            workerCount = nativeQueryRunnerParameters.workerCount.orElse(4);
+
+            storageFormat = DEFAULT_STORAGE_FORMAT;
+            cacheMaxSize = 0;
+            extraProperties = Collections.emptyMap();
+            remoteFunctionServerUds = Optional.empty();
```

Comment:
Make these default values on the fields.  If you put this here because of nativeQueryRunnerParameters, just make that a field too (even if we don't use it to build).